libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -287,4 +393,63 @@ export class OrchestratorAdaptiveRecognizer
             this._resolver = OrchestratorAdaptiveRecognizer.orchestrator.createLabelResolver(snapshot);
         }
     }
+
+    private async tryScoreEntities(text: string, recognizerResult: RecognizerResult) {
+        if (!this.scoreEntities) {
+            return;
+        }
+
+        const results = await this._resolver.score(text, LabelType.Entity);
+        if (!results) {
+            throw new Error(`Failed scoring entities for: ${text}`);
+        }
+
+        // Add full entity recognition result as a 'result' property
+        recognizerResult[this.entityProperty] = results;
+        if (results.length) {
+            if (recognizerResult.entities === null) {
+                recognizerResult.entities = {};
+            }
+
+            results.forEach((result: Result) => {
+                const entityType = result.label.name;
+
+                // add value
+                let values: any[] = recognizerResult.entities[entityType];
+                if (!values) {
+                    values = recognizerResult.entities[entityType] = [];
+                }
+
+                const span = result.label.span;
+                const entityText = text.substr(span.offset, span.length);
+                values.push({
+                    type: entityType,
+                    score: result.score,
+                    text: entityText,
+                    start: span.offset,
+                    end: span.offset + span.length   
+                });               
+
+                // get/create $instance
+                let instanceRoot: any = recognizerResult.entities['$instance'];
+                if (!instanceRoot) {
```

Comment:
`const instanceRoot = recognizerResult.entities['$instance'] ?? {};`

Also, `any` should almost always be avoided.

Reply:
ok